pilot/pkg/serviceregistry/serviceentry/servicediscovery.go

```diff
@@ -64,6 +64,11 @@ type configKey struct {
 	namespace string
 }
 
+type ipWithNetDomain struct {
+	ip      string
+	network string
```

Comment:
I think users can create two wles within the same network and same ip, so network can not distinguish two wles._

Reply:
Oh! What is the use case for that?

Reply:
I believe this the way how @andrewshaoyu  is using

Reply:
But it does not outline the usecase
> different workloadentries may use the same IP, associated with different ServiceEntries with different hosts in different namespaces.

It looks vague. One use case I can think of where same IP is associated with different SE in different hosts in different namespaces is when the VM hosts two services in two different ports. But those two services are in different namespaces in the cluster. So two different WEs are needed with same IP (but different port) in respective namespaces because AFAIK we cannot select an endpoint in another namespace.

Consider this example
```
Service-1: app-1.ns-1.svc.cluster.local, selector: { app: app-1 }, ports: { http: 8080 }
Service-2: app-2.ns-2.svc.cluster.local, selector: { app: app-2 }, ports: { tcp: 9090 }

VM has endpoints related to app-1 and app-2. Its address is 10.0.10.20 in some network
Without loss of generality (and simplicity), we can assume that service and target ports are the same. 
The analysis does not change much if they are different.

So VM: { ip: 10.0.10.20 }, ports { http: 8080, tcp: 9090 }

So 
WE1 called vm-1, with address: 10.10.10.20, port: { http: 8080 } in ns-1 with label {app: app-1}
WE2 called vm-2, with address: 10.10.10.20, port: { tcp: 9090 } in ns-2 with label {app: app-2}
```
Hence we ended up with two Workload Entries with same IP in same network in different namespaces with different exposed ports as a VM hosts two apps. @andrewshaoyu - Is this the usecase you had in mind?

@hzxuzhonghu - Yup. If the above is the use case, I need to come up with a different key for index rather than using `IP+L3Domain`